lib/problems.rb

```diff
@@ -1,13 +1,59 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(n) where n is the length
+# Space Complexity: 0(n)
 def balanced(string)
```

Comment:
Did you know that you can use a hash here where the keys are the open parens and the values are the matching close braces.  That way you can quickly check to see if they match.

This does work however.